### PR TITLE
Verify databus still works with unobtrusive mode

### PR DESCRIPTION
**Pull request by danielmarbach**
We had several issues around unobtrusive conventions. This verifies whether the data bus still works with unobtrusive and "optimized" assembly scanning with v6.

@Particular/nservicebus-maintainers please review and merge

### CODE REVIEW

File: src/NServiceBus.AcceptanceTests/DataBus/When_sending_databus_properties_with_unobtrusive.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        public class Sender : EndpointConfigurationBuilder
+        {
+            public Sender()
+            {
+                EndpointSetup<DefaultServer>(builder =>
+                {
+                    builder.Conventions()
+                        .DefiningCommandsAs(t => t.Namespace != null && t.FullName == typeof(MyMessageWithLargePayload).FullName)
+                        .DefiningDataBusPropertiesAs(t => t.Name.Contains("Payload"));
+                    builder.UseDataBus<FileShareDataBus>().BasePath(@".\databus\sender");
```
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/DataBus/When_sending_databus_properties_with_unobtrusive.cs]: is there any cleanup we need to perform after the test run?

Comment (danielmarbach) [inline src/NServiceBus.AcceptanceTests/DataBus/When_sending_databus_properties_with_unobtrusive.cs]: Existing tests don't do it and databus cleanup seems to be broken by design. So I would say other discussion?

File: src/NServiceBus.AcceptanceTests/DataBus/When_sending_databus_properties_with_unobtrusive.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+                    return Task.FromResult(0);
+                }
+            }
+
+            public class Mutator : IMutateIncomingTransportMessages
+            {
+                public Task MutateIncoming(MutateIncomingTransportMessageContext context)
+                {
+                    if (context.Body.Length > PayloadSize)
```
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/DataBus/When_sending_databus_properties_with_unobtrusive.cs]: what's the intention of this mutator?

Comment (danielmarbach) [inline src/NServiceBus.AcceptanceTests/DataBus/When_sending_databus_properties_with_unobtrusive.cs]: It is an additional assert essentially. Happy to remove it. Copy paste from the previous test

Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/DataBus/When_sending_databus_properties_with_unobtrusive.cs]: let's remove it, I don't see any value from that and it only complicates the test

**Comment by danielmarbach** [comment]
@timbussmann changed

**Pull request merged**